binderhub/repoproviders.py

```diff
@@ -64,6 +66,36 @@ def get_build_slug(self):
         return '{user}-{repo}'.format(user='Rick', repo='Morty')
 
 
+class GitRepoProvider(RepoProvider):
+    """Bare bones git repo provider. Users must provide a resolve_ref in the following form
+
+    http://hostname.com/path/repo_path#sha1
+    """
+
+    name = Unicode("git")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        parsed = urllib.parse.urlparse(self.spec)
```

Comment:
I think there's going to need to be some help here, because a full URL in a URL path is going to be a little tricky, with escapes for `://`, etc. I think we will need to be a bit more explicit, such as specifying that the scheme is the first part, e.g.

    /v2/git/https/github.com/binder-examples/requirements.git@abcdef...

Reply:
yeah so currently i just escape everything and it works for me